### PR TITLE
Improve mock usage

### PR DESCRIPTION
**Pull request by jordisala1991**
This is follow up for: #6421

### CODE REVIEW

**Comment by jordisala1991** [comment]
I didn't change the Container part, because that require a bit more of refactoring, but everything else pointed out here is done: https://github.com/sonata-project/SonataAdminBundle/pull/6421#pullrequestreview-497070158

File: tests/Twig/Extension/SonataAdminExtensionTest.php [hunk 1]
```diff
@@ -2968,10 +2963,20 @@ public function testCanonicalizedLocaleForMoment(?string $expected, string $orig
 
     public function testIsGrantedAffirmative(): void
     {
-        $this->assertTrue(
-            $this->twigExtension->isGrantedAffirmative(['foo', 'bar'])
-        );
-        $this->assertTrue($this->twigExtension->isGrantedAffirmative('foo'));
+        $this->securityChecker->method('isGranted')->willReturnCallback(static function (string $role): bool {
```
Comment (jordisala1991) [inline tests/Twig/Extension/SonataAdminExtensionTest.php:2966]: I copied what you had in your PR @franmomu

**Review by VincentLanglet — APPROVED** [review]

File: tests/Twig/Extension/SonataAdminExtensionTest.php [hunk 2]
```diff
@@ -2968,10 +2963,20 @@ public function testCanonicalizedLocaleForMoment(?string $expected, string $orig
...
+            if ('foo' === $role) {
+                return false;
+            }
+
+            if ('bar' === $role) {
+                return true;
+            }
+
+            return false;
+        });
```
Comment (franmomu) [inline tests/Twig/Extension/SonataAdminExtensionTest.php:2976]: It's way cleaner your way:
```suggestion
       $this->securityChecker->method('isGranted')->willReturnMap([
            ['foo', null, false],
            ['bar', null, true],
        ]);
```

File: tests/Twig/Extension/SonataAdminExtensionTest.php [hunk 3]
```diff
@@ -2963,17 +2963,10 @@ public function testCanonicalizedLocaleForMoment(?string $expected, string $orig
...
-            }
-
-            return false;
-        });
+       $this->securityChecker->method('isGranted')->willReturnMap([
+            ['foo', null, false],
+            ['bar', null, true],
+        ]);
 
         $this->assertTrue($this->twigExtension->isGrantedAffirmative(['foo', 'bar']));
```
Comment (phansys) [inline tests/Twig/Extension/SonataAdminExtensionTest.php:2971]: IMO, we should add some expectations for this method passing the arguments 2 and 3.

**Review by phansys — APPROVED** [review]

**Review by VincentLanglet — DISMISSED** [review]

**Review by phansys — APPROVED** [review]

File: tests/Twig/Extension/SonataAdminExtensionTest.php [hunk 4]
```diff
@@ -2968,10 +2963,13 @@ public function testCanonicalizedLocaleForMoment(?string $expected, string $orig
 
     public function testIsGrantedAffirmative(): void
     {
-        $this->assertTrue(
-            $this->twigExtension->isGrantedAffirmative(['foo', 'bar'])
-        );
-        $this->assertTrue($this->twigExtension->isGrantedAffirmative('foo'));
+       $this->securityChecker->method('isGranted')->willReturnMap([
```
Comment (greg0ire) [inline tests/Twig/Extension/SonataAdminExtensionTest.php:2966]:
```suggestion
        $this->securityChecker->method('isGranted')->willReturnMap([
```

**Review by VincentLanglet — APPROVED** [review]

**Review by franmomu — APPROVED** [review]

**Review by greg0ire — APPROVED** [review]

**Pull request merged**